Hunting Queries/SecurityAlert/WebShellCommandAlertEnrich.yaml

```diff
@@ -0,0 +1,69 @@
+id: d2e6f31b-add1-4f44-b54d-1975a5605c1d
+name: Web shell command alert enrichment
+description: |
+  'Extracts MDATP Alerts that indicate a command was executed by a web shell. Uses time window based querying to idneitfy the potential web shell location on the server, then enriches with Attacker IP and User Agent'
+requiredDataConnectors:
+  - connectorId: AzureSecurityCenter
+    dataTypes:
+      - SecurityAlert
+  - connectorId: MicrosoftCloudAppSecurity
+    dataTypes:
+      - SecurityAlert
+  - connectorId: AzureMonitor(IIS)
+    dataTypes:
+      - W3CIISLog
+tactics:
+  - Privilege Escalation
+  - Persistence
+query: |
+  let timeRange = 3d;  
+  let lookupWindow = 5min;  
+  let lookupBin = lookupWindow / 2.0;
+  let scriptExtensions = dynamic([".php", ".jsp", ".js", ".aspx", ".asmx", ".asax", ".cfm", ".shtml"]);
+  //Extract all MDATP security alert data for alerts associated with command execution by a web shell
+  let alldata = SecurityAlert  
```

Comment:
consider using materialize() - example - let alldata= materialize( SecurityAlert...);  This allows for caching the query which can make later re-use more performant.  Not required.

Reply:
I have used this, really helpful feedback, I hadn't encountered it before so thanks for the tip!